odoo/models.py

```diff
@@ -2983,7 +2983,21 @@ def _field_to_sql(self, alias: str, fname: str, query: (Query | None) = None, fl
             # the result of current sql_field might be 'null'::jsonb
             # ('null'::jsonb)::text == 'null'
             # ('null'::jsonb->>0)::text IS NULL
-            return SQL('(%s->>0)::%s', sql_field, SQL(field._column_type[1]))
+            sql_field = SQL('(%s->>0)::%s', sql_field, SQL(field._column_type[1]))
+
+            if field.type == 'many2one':
+                sql_field = SQL(
+                    '''(
+                        SELECT %(cotable_alias)s.id
+                        FROM %(cotable)s AS %(cotable_alias)s
+                        WHERE %(cotable_alias)s.id = %(ref)s
+                        LIMIT 1
+                    )''',
+                    cotable=SQL.identifier(self.env[field.comodel_name]._table),
+                    cotable_alias=SQL.identifier(Query.make_alias(self.env[field.comodel_name]._table, 'exists')),
+                    ref=sql_field,
+                )
```

Comment:
What about making a left join instead?
```suggestion
            if field.type == 'many2one':
                comodel = self.env[field.comodel_name]
                coalias = query.make_alias(comodel._table, field.name)
                sql_value = sql_field
                sql_field = SQL.identifier(coalias, 'id')
                query.add_join('LEFT JOIN', coalias, comodel._table, SQL("%s = %s", sql_field, sql_value))
```

Reply:
I was thinking about this. But
1. subquery make the sql easier to read
2. performance are the same since both of them don't really JOIN tables but query on the index
3. without using `query.add_join`, accounting team can easily use `_field_to_sql` for company dependent fields to compose the final SQL object for report

Reply:
If the performance is the same, I find the left join easier to read (the code above, but also in the SQL). Now, I understand it is more work for accounting.